File: Interview Prep/Print the index of last non zero element of the array /LastNonZeroIndex.java
import java.util.Scanner;

public class LastNonZeroIndex {
  public static int findLastNonZeroIndex(int[] arr) {
    for (int i = arr.length - 1; i >= 0; i--) {
      if (arr[i] != 0) {
        return i;
      }
    }
    return -1;
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int tc = scanner.nextInt();
    while (tc-- > 0) {
      int n = scanner.nextInt();
      int[] arr = new int[n];
      for (int i = 0; i < n; i++) {
        arr[i] = scanner.nextInt();
      }
      int index = findLastNonZeroIndex(arr);
      System.out.println(index);
    }
    scanner.close();
  }
}

File: Interview Prep/Print the index of last non zero element of the array /LastNonZeroIndex.cpp
#include <iostream>
#include <vector>
using namespace std;

int findLastNonZeroIndex(const vector<int>& arr) {
  for (int i = arr.size() - 1; i >= 0; --i) {
    if (arr[i] != 0) {
      return i;
    }
  }
  return -1;
}

int main() {
  int tc;
  cin >> tc;
  while (tc-- > 0) {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
      cin >> arr[i];
    }
    int index = findLastNonZeroIndex(arr);
    cout << index << endl;
  }
  return 0;
}
